### PR TITLE
feat: add mu03 support

### LINKED ISSUE #98: UI support for MU03 on test nets
**Description**
For MU03 the UI needs to be updated to support new exchanges. 
Namely, USDC/cBRL, USDC/cEUR and EUROC/cEUR

**Acceptance Criteria**
- [x] UI allows users to swap via the new exchanges on Baklava
- [x] UI allows users to swap via the new exchanges on Alfajores

### PR DESCRIPTION
**Pull request by denviljclarke**
### Description

This PR adds support for MU03 changes, it also lays the groundwork to make future stable additions more scaleable. The main changes are: 

- axlEUROC has been added
- The exchanges have been updated based on the config produced by the Mento SDK. 
- The token selection validation logic updated to a more scalable solution. Instead of relying on a case-by-case custom validation, the new approach is based on the exchange configuration to check if pairs are swappable. This will ensure greater flexibility and scalability for future changes.

The deployed preview for this branch can be viewed [here](https://mento-8ijde4480-mentolabs.vercel.app/?vercelToolbarCode=msZTvGphOvMZTxJ)

### Other changes

Token outputs are now limited to valid options, only showing users valid options. When a token is selected only the list of token which are valid to swap are shown instead of showing invalid options and programatically changing their selections. 

I noticed unwanted behaviour when connected to a non Celo network, so I added non Celo network detection as part of logic which handles enabling and disabling of swap button. 

### Tested

I performed the regression tests to confirm core functionality still works
I also selected selected each token as input to ensured only swappable outputs were displayed

### Related issues

- Fixes #98 

### Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] The PR title follows the [conventions](https://www.notion.so/Git-Branching-and-Commit-Message-Conventions-18f66f7d06444cfcbac5725ffbc7c04a?pvs=4#9355048863c549ef92fe210a8a1298aa)
- [x] I have run the [regression tests](https://www.notion.so/Mento-Web-App-Regression-Tests-37bd43a7da8d4e38b65993320a33d557)

### CODE REVIEW

File: src/config/tokens.ts [hunk 1]
```diff
@@ -106,9 +120,34 @@ export function isNativeStableToken(tokenId: string) {
...
+    .filter((tkn) => token !== tkn)
+}
+
 export function getTokenOptionsByChainId(chainId: ChainId): TokenId[] {
   const tokensForChain = TokenAddresses[chainId]
-  return tokensForChain ? (Object.keys(TokenAddresses[chainId]) as TokenId[]) : []
+
+  return tokensForChain
+    ? Object.entries(tokensForChain)
+        .filter(([, tokenAddress]) => tokenAddress !== '') // Allows incomplete 'TokenAddresses' list i.e When tokens are not on all chains
```
Comment (ninocomputer) [inline src/config/tokens.ts:148]: we should probably not allow incomplete tokenAddresses right

File: src/features/swap/SwapForm.tsx [hunk 2]
```diff
@@ -244,15 +269,42 @@ function SlippageRow() {
...
+      toast.error('Could not switch network, does wallet support switching?')
+    }
+  }
 
   const { errors, touched } = useFormikContext<SwapFormValues>()
   const error =
     touched.amount && (errors.amount || errors.fromTokenId || errors.toTokenId || errors.slippage)
-  const text = error ? error : isAccountReady ? 'Continue' : 'Connect Wallet'
+  const text = error
+    ? error
```
Comment (ninocomputer) [inline src/features/swap/SwapForm.tsx:299]: I feel like this is a bit hard to read

File: src/config/tokens.ts (hunk 1 above)
Comment (denviljclarke) [reply src/config/tokens.ts:148]: tokenAddresses (the token list which is used to populate the UI) isn't incomplete. The TokenAddress token to chain map above is incomplete (as there's no EUROC on main net). This is what allows us to move forward and add tokens for a chain without requiring it on all chains

File: src/features/swap/SwapForm.tsx (hunk 2 above)
Comment (denviljclarke) [reply src/features/swap/SwapForm.tsx]: Fixed

**Comment by denviljclarke** [comment]
> some questions. also, wondering how compatible this would be with the recent changes I made to swapForm

There may be some conflict in the submit button, but I've updated the onClick assignment logic to make it easy to add your additional logic as I think this may be merged first so we can use the UI for MU03 testing

**Review by bayological — APPROVED** [review]

**Review by bayological — CHANGES_REQUESTED** [review]
Let's refactor to use the SDK method [getTradeablePairs](https://github.com/mento-protocol/mento-sdk/blob/233ff15b276cae172fd542f83bc4128d8c0b6ec1/src/mento.ts#L106) to avoid hardcoding these things

File: src/features/swap/SwapForm.tsx [hunk 3]
```diff
@@ -244,15 +269,52 @@ function SlippageRow() {
...
+  const dispatch = useAppDispatch()
+  const { errors, touched } = useFormikContext<SwapFormValues>()
+
   const isAccountReady = address && isConnected
+  const isOnCelo = chains.some((chn) => chn.id === chain?.id)
+
+  const switchToNetwork = async () => {
+    try {
+      if (!switchNetworkAsync) throw new Error('switchNetworkAsync undefined')
+      logger.debug('Resetting and switching to Celo')
```
Comment (nvtaveras) [inline src/features/swap/SwapForm.tsx:284]: should we remove this?

Comment (denviljclarke) [reply src/features/swap/SwapForm.tsx:284]: It don't show by default in production so it's okay

**Comment by denviljclarke** [comment]
> Let's refactor to use the SDK method [getTradeablePairs](https://github.com/mento-protocol/mento-sdk/blob/233ff15b276cae172fd542f83bc4128d8c0b6ec1/src/mento.ts#L106) to avoid hardcoding these things

I agree, but this change will require a bit more time. We can merge this and I'll create another issue for full SDK integration.

**Review by nvtaveras — APPROVED** [review]

**Pull request merged**